src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -26,13 +26,14 @@ public void Add(JobDefinition jobDefinition)
     {
         AssertNoDuplicateJobNames(jobDefinition.CustomName);
 
-        var isTypeUpdate = allJobs.Any(j => j.JobFullName == jobDefinition.JobFullName);
-        if (isTypeUpdate)
+        if (!allJobs.Add(jobDefinition))
         {
-            Remove(jobDefinition);
+            throw new InvalidOperationException(
+                """
+                Job registration conflict detected. Another job with the same type, parameters, or cron expression already exists.
```

Comment:
If possible, add here some metadata about the JobDefinition (type, cron, parameter)

Reply:
Added the job type. CRON expression might not be set anyway.